spec/ttt/game_rules_spec.clj

```diff
@@ -0,0 +1,53 @@
+(ns ttt.game-rules-spec
+  (:require [speclj.core :refer [describe context it should=]]
+            [ttt.game-rules :as rules]))
+
+
+(describe "Game Rules"
+  (context "#win?"
```

Comment:
_food for thought; no action required_

The discussion topic Tony started is among my favorites: what is the minimum set of tests required that:

* fully _exercises_ the behavior of the production code (i.e. you drove every expression in the production code until you couldn't write any more tests)
* fully _describes_ the behavior of the production code (i.e. the text in the `it` clauses reads well to human beings, if you were to remove all the test code)

I like to think about this as if my tests are teaching the production code to do progressively more difficult things, like:

* always return false (given a game board with no winner)
* learn to look for a specific player X in a specific row (the first row)
* learn to iterate over all rows by checking for a winner in the 2nd row
* learn about another player, O, by looking for O to win in any row (it already knows how to check rows, at this point)
* learn to check the first column, then all columns, by checking for a win in the first and second columns
* learn to check both diagonals, by checking for a win in each diagonal
* if necessary, make sure a row, column, or diagonal that is occupied by multiple players returns false (depends on whether the production code was able to get away with a lesser check, up to this point)

There are a number of ways to go about adding/modifying tests to go through enough steps to make your production code do everything it needs to do.  I think it's kind of neat to think about what is the minimal set of tests that still meets the exercise/describe criteria above, but I wouldn't get too carried away looking for a perfect solution.

The bottom line: Keep writing tests until the production code does everything it should, and try to keep them organized in some fashion that seems logical to humans, as you go.  If you wind up with a few extra tests (within reason), it's no big deal.